run_demo.sh

```diff
@@ -40,33 +42,96 @@ function check_hostname(){
   fi
 }
 
-if [ -z "${1:-}" ]; then
-  echo "Usage: $0 <diracx_src_dir> [other source directories]"
-  exit 1
-fi
-diracx_repo_dir="$(readlink -f "${1}")"
-diracx_src_dir="${diracx_repo_dir}/src/diracx"
-if [[ ! -d "${diracx_src_dir}" ]]; then
-  printf "\U26A0\UFE0F Error: %s is not a clone of DiracX!" "${diracx_repo_dir}"
-  exit 1
-fi
+usage="${0##*/} [-h|--help] [--exit-when-done] [--] <diracx_src_dir> [other source directories]"
+
+# Parse command-line switches
+exit_when_done=0
+while [ -n "$1" ]; do case $1 in
+	# Print a brief usage summary and exit
+	-h|--help|-\?)
+		printf 'Usage: %s\n' "$usage"
+		exit ;;
+
+	# # Unbundle short options
+	# -[niladic-short-opts]?*)
+	# 	tail="${1#??}"
+	# 	head=${1%"$tail"}
+	# 	shift
+	# 	set -- "$head" "-$tail" "$@"
+	# 	continue ;;
+
+	# # Expand parametric values
+	# -[monadic-short-opts]?*|--[!=]*=*)
+	# 	case $1 in
+	# 		--*) tail=${1#*=}; head=${1%%=*} ;;
+	# 		*)   tail=${1#??}; head=${1%"$tail"} ;;
+	# 	esac
+	# 	shift
+	# 	set -- "$head" "$tail" "$@"
+	# 	continue ;;
```

Comment:
Is this expected?

Reply:
Yeah we currently don't have any `niladic` or `monadic` options but I think this is fiddley enough to write in portable bash that it's worth having commented out in preparation :)